### PR TITLE
feature/IESP-214

### PR DESCRIPTION
**Pull request by luigifortunato00**
Added CompletableFuture to make asynchronous calls

### CODE REVIEW

File: src/main/java/it/pagopa/interop/probing/scheduler/scheduler/ScheduledTasks.java [hunk 1]
```diff
@@ -32,27 +33,35 @@ public class ScheduledTasks {
...
         "- [CID= " + UUID.randomUUID().toString().toLowerCase() + "]");
     logger.logSchedulerStart();
     try {
-      Integer offset = 0;
+      offset = 0;
       while (true) {
-        PollingEserviceResponse response =
-            eserviceService.getEservicesReadyForPolling(limit, offset);
-        for (EserviceContent service : response.getContent()) {
+        CompletableFuture<PollingEserviceResponse> response = CompletableFuture
```
Comment (nttdata-rtorsoli) [inline src/main/java/it/pagopa/interop/probing/scheduler/scheduler/ScheduledTasks.java:47]: It's all right , but it is better to create an proper ExecutionContext for non blocking operations such as newFixedThreadPool.
At this proposite please get in touch with your colleague that are using Spring @Async, it is the same thing but for readability it is better to use the same  pattern.

**Review by nttdata-rtorsoli — APPROVED** [review]
See comments

File: src/main/java/it/pagopa/interop/probing/scheduler/scheduler/ScheduledTasks.java [hunk 2]
```diff
@@ -32,27 +33,35 @@ public class ScheduledTasks {
...
-            eserviceService.updateLastRequest(service.getEserviceRecordId(), ChangeLastRequest
-                .builder().lastRequest(OffsetDateTime.now(ZoneOffset.UTC)).build());
-            servicesSend.sendMessage(service);
+            CompletableFuture.runAsync(() -> {
+              try {
+                eserviceService.updateLastRequest(service.getEserviceRecordId(), ChangeLastRequest
+                    .builder().lastRequest(OffsetDateTime.now(ZoneOffset.UTC)).build());
+                servicesSend.sendMessage(service);
+              } catch (Exception e) {
+                e.printStackTrace();
```
Comment (galales) [inline src/main/java/it/pagopa/interop/probing/scheduler/scheduler/ScheduledTasks.java:57]: please add proper log for the error

File: src/main/java/it/pagopa/interop/probing/scheduler/scheduler/ScheduledTasks.java [hunk 3]
```diff
@@ -32,27 +33,35 @@ public class ScheduledTasks {
...
+
 
   @Scheduled(cron = "${scheduler.cron.expression}")
   public void scheduleFixedDelayTask() {
     MDC.put(LoggingPlaceholders.TRACE_ID_PLACEHOLDER,
         "- [CID= " + UUID.randomUUID().toString().toLowerCase() + "]");
     logger.logSchedulerStart();
     try {
-      Integer offset = 0;
+      offset = 0;
```
Comment (galales) [inline src/main/java/it/pagopa/interop/probing/scheduler/scheduler/ScheduledTasks.java:45]: shouldn't `offset` declaration be kept in the method scope?

File: src/main/java/it/pagopa/interop/probing/scheduler/scheduler/ScheduledTasks.java [hunk 4]
```diff
@@ -40,22 +44,29 @@ public void scheduleFixedDelayTask() {
     logger.logSchedulerStart();
     try {
       Integer offset = 0;
+      PollingEserviceResponse response = PollingEserviceResponse.builder().build();
```
Comment (galales) [inline src/main/java/it/pagopa/interop/probing/scheduler/scheduler/ScheduledTasks.java:47]: does this init have a purpose considering that `response` value is replaced right away?

**Review by galales — APPROVED** [review]

File: src/main/java/it/pagopa/interop/probing/scheduler/scheduler/ScheduledTasks.java (hunk 4 above)
Comment (luigifortunato00) [reply src/main/java/it/pagopa/interop/probing/scheduler/scheduler/ScheduledTasks.java]: Done

**Review by nttdata-rtorsoli — APPROVED** [review]

**Review by galales — APPROVED** [review]

**Pull request merged**